Library/Homebrew/dev-cmd/determine-test-runners.rb

```diff
@@ -0,0 +1,119 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "cli/parser"
+require "test_runner_formula"
+require "github_runner_matrix"
+
+module Homebrew
+  extend T::Sig
+
+  sig { returns(Homebrew::CLI::Parser) }
+  def self.determine_test_runners_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `determine-test-runners` <testing-formulae> [<deleted-formulae>]
+
+        Determines the runners used to test formulae or their dependents.
+      EOS
+      switch "--eval-all",
+             description: "Evaluate all available formulae, whether installed or not, to determine testing " \
+                          "dependents."
+      switch "--dependents",
+             description: "Determine runners for testing dependents. Requires `--eval-all` or `HOMEBREW_EVAL_ALL`."
+
+      named_args min: 1, max: 2
+
+      hide_from_man_page!
+    end
+  end
+
+  sig {
+    params(
+      version:          String,
+      arch:             Symbol,
+      ephemeral:        T::Boolean,
+      ephemeral_suffix: T.nilable(String),
+    ).returns(T::Hash[Symbol, T.any(String, T::Boolean)])
+  }
+  def self.runner_spec(version, arch:, ephemeral:, ephemeral_suffix: nil)
+    case arch
+    when :arm64 then { runner: "#{version}-arm64#{ephemeral_suffix}", clean: !ephemeral }
+    when :x86_64 then { runner: "#{version}#{ephemeral_suffix}", clean: !ephemeral }
+    else raise "Unexpected arch: #{arch}"
+    end
+  end
+
+  sig { void }
+  def self.determine_test_runners
+    args = determine_test_runners_args.parse
+
+    eval_all = args.eval_all? || Homebrew::EnvConfig.eval_all?
+
+    odie "`--dependents` requires `--eval-all` or `HOMEBREW_EVAL_ALL`!" if args.dependents? && !eval_all
+
+    Formulary.enable_factory_cache!
+
+    testing_formulae = args.named.first.split(",")
+    testing_formulae.map! { |name| TestRunnerFormula.new(Formulary.factory(name), eval_all: eval_all) }
+                    .freeze
+    deleted_formulae = args.named.second&.split(",")
+
+    linux_runner       = ENV.fetch("HOMEBREW_LINUX_RUNNER") { raise "HOMEBREW_LINUX_RUNNER is not defined" }
```

Comment:
```suggestion
    linux_runner       = ENV.fetch("HOMEBREW_LINUX_RUNNER")
```

as this will already raise. I think we have helpers already that provide a better error message if that's what you're looking for.